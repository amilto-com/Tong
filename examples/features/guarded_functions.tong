// Demonstration of multi-clause guarded functions
// Define a sign function using guarded clauses
fn sign(x) if x > 0 {
  print("positive")
  return 1
}
fn sign(x) if x == 0 {
  print("zero")
  return 0
}
fn sign(x) if x < 0 {
  print("negative")
  return -1
}

print(sign(5))
print(sign(0))
print(sign(-7))

// Factorial with guarded base case
fn fact(n) if n == 0 { return 1 }
fn fact(n) if n > 0 { return n * fact(n - 1) }

print(fact(5))  // 120
