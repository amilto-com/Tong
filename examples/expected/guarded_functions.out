positive 1 zero 0 negative -1 120